packages/core/src/DdSdkReactNative.tsx

```diff
@@ -38,7 +47,25 @@ export class DdSdkReactNative {
             configuration.additionalConfig[DdSdkReactNative.DD_SOURCE_KEY] = 'react-native';
             configuration.additionalConfig[DdSdkReactNative.DD_NATIVE_VIEW_TRACKING_KEY] = configuration.nativeViewTracking;
             if (configuration.verbosity != undefined) {
-                configuration.additionalConfig[DdSdkReactNative.DD_SDK_VERBOSITY_KEY] = configuration.verbosity
+                configuration.additionalConfig[DdSdkReactNative.DD_SDK_VERBOSITY_KEY] = configuration.verbosity;
+            }
+
+            if (configuration.proxyConfig) {
+                const additionalConfig = configuration.additionalConfig;
+                const proxyConfig = configuration.proxyConfig;
+
+                additionalConfig[DdSdkReactNative.DD_PROXY_TYPE_KEY] = proxyConfig.type;
+                additionalConfig[DdSdkReactNative.DD_PROXY_ADDRESS_KEY] = proxyConfig.address;
+                additionalConfig[DdSdkReactNative.DD_PROXY_PORT_KEY] = proxyConfig.port;
+                if (proxyConfig.username && proxyConfig.password) {
+                    if (proxyConfig.type == ProxyType.SOCKS) {
+                        console.warn("SOCKS proxy configuration doesn't support Basic authentication out"
+                            + " of the box, please set it up using java.net.Authenticator by yourself.")
```

Comment:
This warning is specific to Android. it could say something like `SOCKS proxy configuration doesn't support Basic authentication out of the box, please refer to platform documentations.`

Reply:
What is the final decision regarding iOS? Is it possible to support SOCKS+auth there? I'm afraid just saying about platform documentation is not enough, at least for Android, because solution is not obvious considering we use 3rd party networking library.

Reply:
No, it is not supported on iOS. So we might consider mentioning that in the warning.

Reply:
Also this is misleading since they can't use a Java class from javascript. It's not just that they need to implement their own Authenticator, they actually need to initialise the Android SDK (not even the bridge) from the native side. 
Let's simply say that SOCKS proxy with authentication is not supported at all.

Reply:
There is no need to initialise Android SDK from native side in case of implementing `SOCKS` + auth scheme for Android, user should just do `static java.net.Authenticator.setDefault(....)` from native side and it can be done at any time (just before 1st call is made by Android SDK to the server, which will happen only after some time) and doesn't depend on the Android SDK code. SOCKS layer implementation then just later will call `static java.net.Authenticator.requestPasswordAuthentication(...)` when needed.

However, considering it is not working on iOS (at least currently) it means customers will be unable to use this scheme for Android as well (considering they use the same server for both platforms), so I'm ok saying it is not working. But that means there is no need to pass credentials to the native code in this case, as requested by @maxep earlier. Or we pass them anyway, even if it doesn't work currently? wdyt?